--- src/ga_optim.c	2005-02-24 13:54:19.000000000 +0100
+++ src/ga_optim.new	2014-11-01 19:09:54.427210228 +0100
@@ -47,6 +47,10 @@
 
 #include "gaul/ga_optim.h"
 
+#ifndef max
+	#define max(a,b) ((a>b)?a:b)
+#endif
+
 /*
  * Here is a kludge.
  *
@@ -506,7 +510,7 @@
  * FIXME: It would be more efficient to insert the immigrants correctly.
  */
 #pragma omp parallel for \
-   shared(pops,num_pops) private(current_island) \
+   shared(pops) private(current_island) \
    schedule(static)
   for(current_island=0; current_island<num_pops; current_island++)
     {
